samples/subsys/shell/shell_module/README.rst

```diff
@@ -0,0 +1,157 @@
+various commands exposed through the Kconfig's.
+
+Requirements
+************
+
+* A target configured with the shell interface, exposed through any of its backends.
+
+Building and Running
+********************
+
+This sample can be found under :zephyr_file:`samples/subsys/shell/shell_module`
+in the Zephyr tree.
+
+The sample can be built for several platforms.
+
+Emulation Targets
+=================
+
+The sample may run on emulation targets. The following commands build the
+application for the qemu_x86.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: qemu_x86
+   :goals: run
+   :compact:
+
+After running the application, the console displays the shell interface, and
+shows the shell prompt, at which point the user may start the interaction.
+
+On-Hardware (Default Backend)
+=============================
+
+The application can also be built for boards on their default Shell
+backend (usually UART). The following commands are applicable to target the
+nrf52840dk/nrf52840.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :compact:
+
+Build with GPIO-related commands
+--------------------------------
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :gen-args: -DCONFIG_GPIO=y -DCONFIG_GPIO_SHELL=y
+   :compact:
+
+Build with I2C-related commands
+-------------------------------
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :gen-args: -DCONFIG_I2C=y -DCONFIG_I2C_SHELL=y
+   :compact:
+
+Build with Sensor-related commands
+----------------------------------
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :gen-args: -DCONFIG_SENSOR=y -DCONFIG_SENSOR_SHELL=y
+   :compact:
+
+Backend - USB CDC ACM
+=====================
+
+The application may be configured to expose Shell over USB CDC ACM.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :gen-args: -DOVERLAY_CONFIG='overlay-usb.conf' -DDTC_OVERLAY_FILE='usb.overlay'
+   :compact:
+
+Backend - Shell over Bluetooth LE (NUS)
+=======================================
+
+The application may also be built for boards with Bluetooth capabilities, by
+exposing the shell interface over Bluetooth LE (NUS). The following commands
+are applicable to the nrf52840dk/nrf52840.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :gen-args: -DOVERLAY_CONFIG='overlay-bt.conf' -DDTC_OVERLAY_FILE='bt.overlay'
+   :compact:
```

Comment:
mmm shouldn't we just encourage people to use existing snippets and drop these overlays? They seem to be mostly a copy-paste of the equivalent snippets. I think these docs on "how to enable CDC/NUS/RTT shell" should be moved to the actual shell documentation page, as they have nothing to do with this sample specifically (even more so if the generic snippets can be used)

Reply:
I've refactored the Shell documentation to reference the code-snippets (now shell-compatible for both `cdc-acm-console`, as well as for `nus-console`)